pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -101,17 +101,19 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 	modifiedDataVolume := dataVolume.DeepCopy()
 	modified := false
 
-	config, err := wh.cdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
-	if err != nil {
-		return toAdmissionResponseError(err)
-	}
-	if config.Spec.DataVolumeTTLSeconds != nil &&
-		modifiedDataVolume.Annotations[controller.AnnDeleteAfterCompletion] != "false" {
-		if modifiedDataVolume.Annotations == nil {
-			modifiedDataVolume.Annotations = make(map[string]string)
+	if ar.Request.Operation == admissionv1.Create {
+		config, err := wh.cdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+		if err != nil {
+			return toAdmissionResponseError(err)
+		}
+		if config.Spec.DataVolumeTTLSeconds != nil &&
+			modifiedDataVolume.Annotations[controller.AnnDeleteAfterCompletion] != "false" {
+			if modifiedDataVolume.Annotations == nil {
```

Comment:
if `modifiedDataVolume.Annotations[controller.AnnDeleteAfterCompletion]` is not False, then how can the `Annotations` be nil?

Reply:
oh, it is the go way of handling empty/nil Annotations map

Reply:
exactly

Reply:
this is strange to me if modifiedDataVolume.Annotations is nil wouldn't modifiedDataVolume.Annotations[controller.AnnDeleteAfterCompletion] be nil de-referencin and panic?

Reply:
This is an interesting behavior of go, weird :)

Reply:
Yep, you can check a key on a nil map in go with no fear :)
However, if both of you didn't like it, I'll re-write these lines.

Reply:
I am fine with both ways, sometimes it is easier to read with the more explicit version.